python/huggingfaceserver/huggingfaceserver/vllm/vllm_completions.py

```diff
@@ -387,39 +391,62 @@ def _validate_prompt_and_tokenize(
                 f"Please reduce the length of the messages or completion.",
             )
         else:
-            return input_ids
+            return input_ids, input_text
+
+    def _get_decoded_token(self, logprob: Logprob, token_id: int) -> str:
+        if logprob.decoded_token is not None:
+            return logprob.decoded_token
+        return self.tokenizer.decode(token_id)
 
     def _create_logprobs(
         self,
         token_ids: List[int],
-        top_logprobs: Optional[List[Optional[Dict[int, Logprob]]]] = None,
-        num_output_top_logprobs: Optional[int] = None,
+        top_logprobs: List[Optional[Dict[int, Logprob]]],
+        num_output_top_logprobs: int,
         initial_text_offset: int = 0,
     ) -> Logprobs:
         """Create OpenAI-style logprobs."""
-        logprobs = Logprobs(text_offset=[], tokens=[], token_logprobs=[])
+        logprobs = Logprobs(
```

Comment:
vLLM allows [adding 'None' to token_logprobs and top_probs](https://github.com/vllm-project/vllm/blob/5fa6e9876ed8212d74a71a7df7654fd4b94dd659/vllm/entrypoints/openai/protocol.py#L548-L553) while openai client [does not allows `None` values](https://github.com/openai/openai-python/blob/main/src/openai/types/completion_choice.py#L11-L18).

vLLM'sType definition of LogProbs:
```python
class CompletionLogProbs(OpenAIBaseModel):
    text_offset: List[int] = Field(default_factory=list)
    token_logprobs: List[Optional[float]] = Field(default_factory=list)
    tokens: List[str] = Field(default_factory=list)
    top_logprobs: List[Optional[Dict[str,
                                     float]]] = Field(default_factory=list)
```

OpenAI client and KServe's type defintion of LogProbs:
```python
class Logprobs(BaseModel):
    text_offset: Optional[List[int]] = None

    token_logprobs: Optional[List[float]] = None

    tokens: Optional[List[str]] = None

    top_logprobs: Optional[List[Dict[str, float]]] = None
```
vLLM use the 'None' value for special tokens which does not have log probabilities.  For example, In the below completion, top_logprobs and token_logprobs is marked as `null` for special token `</s>`. To bypass the pydantic validation here, I am creating the LogProbs with empty lists and appending the values to the list. Directly assigning the list which contains `None` value will raise pydantic validation error.

```json
"logprobs": {
                "text_offset": [
                    0,
                    4,
                    6,
                    7,
                    9,
                    14,
                    17,
                    21,
                    22,
                    31,
                    32,
                    35,
                    39,
                    44,
                    49,
                    54,
                    61
                ],
                "token_logprobs": [
                    null,
                    -9.352765083312988,
                    -1.4278249740600586,
                    -0.976689338684082,
                    -5.6148481369018555,
                    -4.214991569519043,
                    -4.99854040145874,
                    -5.968787670135498,
                    -11.009231567382812,
                    -3.1531941890716553,
                    -1.4167277812957764,
                    -2.766524314880371,
                    -6.9396467208862305,
                    -1.3619931936264038,
                    -4.619960308074951,
                    -5.6248779296875,
                    -2.2152767181396484
                ],
                "tokens": [
                    "</s>",
                    "Hi",
                    ",",
                    " I",
                    " love",
                    " my",
                    " cat",
                    "-",
                    " Labrador",
                    "!",
                    " He",
                    " has",
                    " tiny",
                    " ears",
                    " with",
                    " fluffy",
                    " white"
                ],
                "top_logprobs": [
                    null,
                    {
                        "Hi": -9.352765083312988,
                        "I": -1.4278708696365356,
                        "The": -2.4365129470825195
                    },
                    {
                        ",": -1.4278249740600586,
                        "!": -1.934173583984375
                    },
                    {
                        " I": -0.976689338684082,
                        " ": -2.723400115966797
                    },
                    {
                        " love": -5.6148481369018555,
                        "'m": -1.015452265739441,
                        " have": -1.9374703168869019
                    },
                    {
                        " my": -4.214991569519043,
                        " your": -1.7619359493255615,
                        " the": -1.999145269393921
                    },
                    {
                        " cat": -4.99854040145874,
                        " new": -3.4642574787139893,
                        " old": -4.73804235458374
                    },
                    {
                        "-": -5.968787670135498,
                        ".": -1.453755497932434,
                        ",": -1.841694951057434
                    },
                    {
                        " Labrador": -11.009231567382812,
                        " I": -1.754422903060913,
                        " she": -3.075488328933716
                    },
                    {
                        "!": -3.1531941890716553,
                        " mix": -1.0394361019134521,
                        ".": -2.1872146129608154
                    },
                    {
                        " He": -1.4167277812957764,
                        "\n": -2.0672662258148193
                    },
                    {
                        " has": -2.766524314880371,
                        "'s": -1.0847474336624146,
                        " is": -1.547521710395813
                    },
                    {
                        " tiny": -6.9396467208862305,
                        " a": -1.3877270221710205,
                        " been": -2.3109371662139893
                    },
                    {
                        " ears": -1.3619931936264038,
                        " paws": -2.2743258476257324
                    },
                    {
                        " with": -4.619960308074951,
                        " and": -0.805719792842865,
                        ",": -1.6155686378479004
                    },
                    {
                        " fluffy": -5.6248779296875,
                        " a": -1.4977400302886963,
                        " tiny": -3.006150484085083
                    },
                    {
                        " white": -2.2152767181396484,
                        " fur": -1.9012728929519653
                    }
                ]
            }
```

Reply:
@gavrissh @yuzisun

Reply:
Based on the open inference protocol discussion we plan to maintain a compatible version of OpenAI protocol, slight changes like this should be ok.